### PR TITLE
DSD-1791: Fix SearchBar select cutoff issue

### PR DESCRIPTION
**Pull request by jackiequach**
Fixes JIRA ticket [DSD-1791](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1791)

The `Select` component is sized to fit the largest option size; if the option exceeds the maxWidth of 255px (~30 characters) then the text is truncated.

## This PR does the following:

- Removes the right margin on the Select component of the SearchBar which caused the longest option text to be truncated even if the Select component is not at the maxWidth.
- Fixes issue on Safari where overflowed text was not being truncated due to not having `overflow: hidden`

## How has this been tested?

- Locally on Storybook (Chrome, Firefox, Safari)

## Accessibility concerns or updates

-

### Accessibility Checklist

- [ ] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [ ] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [ ] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [ ] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [ ] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

-

### Checklist:

- [ ] I have updated the Storybook documentation and changelog accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

[DSD-1791]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/theme/components/searchBar.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ const SearchBar = defineMultiStyleConfig({
       marginBottom: { base: "-1px", md: "0" },
       maxWidth: { base: undefined, md: "255px" },
       textOverflow: "ellipsis",
-      marginRight: { base: undefined, md: "-1px" },
+      overflow: "hidden",
```
Comment (bigfishdesign13) [inline src/theme/components/searchBar.ts:44]: Why was this added? When inspecting the DOM, it's being overridden by some other style.

File: src/components/SearchBar/SearchBar.mdx [hunk 2]
```diff
@@ -100,6 +100,9 @@ page for a full demonstration.
 The `optionsData` prop is an array of objects that contain the `text` and
 `value` properties.
 
+**Note:** when `Select` options have a length greater than ~30 characters then
+the text will be truncated.
```
Comment (bigfishdesign13) [inline src/components/SearchBar/SearchBar.mdx]: Nit: use `approximately` or `approx.` indtead of `~`.

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +19,10 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 - Updates the `Table` component to add the `titleText` and `showTitleText` props to control the `<caption>` element and `aria-label` attribute.
 - Updates the styles for the `Template` component to use `display` and `overflow` for the primary content area to better accommodate the horizontal scrolling in the `Table` component.
 
+### Fixes
+
+- Fixes issue with cut off `Select` options in the `SearchBar` component.
```
Comment (bigfishdesign13) [inline CHANGELOG.md]: Options can still be truncated based on the length of the option title.  Move this to the `Updates` section and say something about allowing longer option titles before truncation.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Needs some adjustments. Also, to improve the spacing within the `Select` component and ultomately allow for longer option titles, we should update the `paddingEnd` style that is being applied to the `Select` component and reduce it from 3rem to 2.5rem.

https://github.com/NYPL/nypl-design-system/blob/f0d931474cbf16361c12ece8e465aae8dd4e4d4f/src/theme/components/select.ts#L27

File: src/theme/components/searchBar.ts (hunk 1 above)
Comment (jackiequach) [reply src/theme/components/searchBar.ts:44]: On Safari, overflowed text was not truncated due to this missing prop.

**Comment by jackiequach** [comment]
@bigfishdesign13 There is no design token for 2.5rem. Should we just hard-code this?

File: src/theme/components/searchBar.ts (hunk 1 above)
Comment (bigfishdesign13) [reply src/theme/components/searchBar.ts:44]: Weird. I had never noticed that before. Good catch.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!
